### ISSUE TITLE
Parsing of URI in processRefNode broken

### ISSUE THREAD

**Issue by thijskh** [robrichards/xmlseclibs#47]
Hi,

processRefNode() contains this code:
```php
        if ($uri = $refNode->getAttribute("URI")) {
            $arUrl = parse_url($uri);
            if (empty($arUrl['path'])) {
...
            } else {
                $dataObject = file_get_contents($arUrl);
```

I think this cannot work. parse_url will either return an array or false.

If it returns false or an array without path, the (omitted) code block after the if is triggered. This can only happen when the URL is very invalid, or `scheme://host` without a path. I'm not sure why those two cases should be united.

In all other cases, the else block will be triggered. However, by then we're sure that `$arUrl` will contain an array with a 'path' element, which when cast to string to be fed to file_get_contents will always be "Array".

So either I'm completely misunderstanding this code, or it cannot work in this form.

**Comment by robrichards** [comment]
Yes that does appear buggy how external resources are handled (or actually aren't able to be handled). I will look into fixing that.

**Comment by thijskh** [comment]
Not sure how crucial it is; I only discovered this by reading the code, not by actually running into the issue.

**Comment by robrichards** [comment]
resolved in v4
